### ISSUE TITLE
Glow on GAS with NaN validation error

### ISSUE THREAD

**Issue by Zardinality** [ikostrikov/pytorch-flows#4]
To reproduce, use:
```bash
python main.py --dataset GAS --flow glow
```
Printed message for first epoch:
```
Warning: Results for GLOW are not as good as for MAF yet.
Warning: Results for GLOW are not as good as for MAF yet.
Warning: Results for GLOW are not as good as for MAF yet.
Warning: Results for GLOW are not as good as for MAF yet.
Warning: Results for GLOW are not as good as for MAF yet.
Train Epoch: 0 [0/852174 (0%)]	Loss: 6.648179
Train Epoch: 0 [100000/852174 (12%)]	Loss: -2.193499
Train Epoch: 0 [200000/852174 (23%)]	Loss: -2.497202
Train Epoch: 0 [300000/852174 (35%)]	Loss: -1.122860
Train Epoch: 0 [400000/852174 (47%)]	Loss: -2.892694
Train Epoch: 0 [500000/852174 (59%)]	Loss: -2.268665
Train Epoch: 0 [600000/852174 (70%)]	Loss: -3.088916
Train Epoch: 0 [700000/852174 (82%)]	Loss: -3.798649
Train Epoch: 0 [800000/852174 (94%)]	Loss: -2.554942

Validation set: Average loss: nan

Best validation at epoch 0: Average loss: inf
```

**Comment by Zardinality** [comment]
After adding a `HalfTanh` exactly like MADE, the validation behaves normal now:
```
Warning: Results for GLOW are not as good as for MAF yet.
Warning: Results for GLOW are not as good as for MAF yet.
Warning: Results for GLOW are not as good as for MAF yet.
Warning: Results for GLOW are not as good as for MAF yet.
Warning: Results for GLOW are not as good as for MAF yet.
Train Epoch: 0 [0/852174 (0%)]	Loss: 6.640622
Train Epoch: 0 [100000/852174 (12%)]	Loss: -2.256141
Train Epoch: 0 [200000/852174 (23%)]	Loss: -2.629832
Train Epoch: 0 [300000/852174 (35%)]	Loss: -1.263118
Train Epoch: 0 [400000/852174 (47%)]	Loss: -2.958904
Train Epoch: 0 [500000/852174 (59%)]	Loss: -2.457846
Train Epoch: 0 [600000/852174 (70%)]	Loss: -3.236554
Train Epoch: 0 [700000/852174 (82%)]	Loss: -4.018288
Train Epoch: 0 [800000/852174 (94%)]	Loss: -2.708811

Validation set: Average loss: 3.0890

Best validation at epoch 0: Average loss: 3.0890
```
Does it need a patch? I can start a pull request.

**Comment by ikostrikov** [comment]
I would add a flag for activations since in the original code the authors use different activations for this dataset:

https://github.com/gpapamak/maf/blob/master/run_experiments.py#L46

**Comment by Zardinality** [comment]
Fixed in #5
